pkg/simulator/ssh.go

```diff
@@ -52,10 +52,10 @@ func (s *Simulator) Attack() error {
 		"BastionIP": bastion,
 	}).Info("Connecting to bastion")
 
-	kp, err := s.StateProvider.GetSSHKeyPair()
+	kp, err := s.SSHStateProvider.GetSSHKeyPair()
 	if err != nil {
 		return errors.Wrap(err, "Error getting SSH keypair")
 	}
 
-	return ssh.SSH(bastion, *kp)
+	return ssh.SSH(bastion, *kp, s.ProgressStateProvider)
```

Comment:
The APi for the SSH module is not very idiomatic.  I made this when I was in the early stage of learning go so everything is functions.  It needs some refactoring to introduce a struct type and convert to methods. This PR wa salready big enough though!